tests/integration/src/test/java/org/mule/transformers/simple/SessionVariableAfterJavaComponentTestCase.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.transformers.simple;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mule.api.transport.PropertyScope.SESSION;
+import org.mule.api.MuleEventContext;
+import org.mule.api.lifecycle.Callable;
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.tck.junit4.rule.SystemProperty;
+import org.mule.util.concurrent.Latch;
+
+
+import org.junit.Rule;
+import org.junit.Test;
+
+public class SessionVariableAfterJavaComponentTestCase extends FunctionalTestCase
+{
+
+    private static final String SESSION_VARIABLE_VALUE = "TEST";
+
+    @Rule
+    public SystemProperty sessionVariableSystemProperty = new SystemProperty("sessionVariable", SESSION_VARIABLE_VALUE);
```

Comment:
Seems to me that the property value can be fixed, why do you need a system property?